src/network/networkd-sriov.c

```diff
@@ -103,3 +107,228 @@ int link_request_sr_iov_vfs(Link *link) {
 
         return 0;
 }
+
+static int find_ifindex_from_pci_dev_port(sd_device *pci_dev, const char *dev_port) {
+        _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
+        sd_device **devices;
+        size_t n_devices;
+        int ifindex, r;
+
+        assert(pci_dev);
+        assert(dev_port);
+
+        r = sd_device_enumerator_new(&e);
+        if (r < 0)
+                return r;
+
+        r = sd_device_enumerator_allow_uninitialized(e);
+        if (r < 0)
+                return r;
+
+        r = sd_device_enumerator_add_match_parent(e, pci_dev);
+        if (r < 0)
+                return r;
+
+        r = sd_device_enumerator_add_match_subsystem(e, "net", true);
+        if (r < 0)
+                return r;
+
+        r = sd_device_enumerator_add_match_sysattr(e, "dev_port", dev_port, true);
+        if (r < 0)
+                return r;
+
+        r = device_enumerator_scan_devices(e);
+        if (r < 0)
+                return r;
+
+        devices = device_enumerator_get_devices(e, &n_devices);
+        if (!devices || n_devices <= 0)
+                return -ENODEV;
+        if (n_devices > 1)
+                return -ENXIO;
+
+        r = sd_device_get_ifindex(devices[0], &ifindex);
+        if (r < 0)
+                return r;
+
+        assert(ifindex > 0);
+        return ifindex;
+}
+
+int link_get_sr_iov_phys_port(Link *link, Link **ret) {
+        _cleanup_(sd_device_unrefp) sd_device *pci_dev = NULL, *pci_physfn_dev = NULL;
+        const char *pci_syspath, *pci_physfn_syspath, *dev_port;
+        _cleanup_closedir_ DIR *dir = NULL;
+        bool found = false;
+        int r;
+
+        assert(link);
+        assert(link->manager);
+
+        if (link->phys_port_ifindex > 0) {
+                r = link_get_by_index(link->manager, link->phys_port_ifindex, ret);
+                if (r < 0)
+                        return r;
+
+                return link->phys_port_ifindex;
+        }
+
+        if (link->phys_port_ifindex < 0) /* Previously parsed, and not found. */
+                return -ENODEV;
+
+        link->phys_port_ifindex = -1;
+
+        if (!link->sd_device)
+                return -ENODEV;
+
+        r = sd_device_get_sysattr_value(link->sd_device, "dev_port", &dev_port);
+        if (r < 0)
+                return r;
+
+        r = sd_device_get_parent_with_subsystem_devtype(link->sd_device, "pci", NULL, &pci_dev);
+        if (r < 0)
+                return r;
+
+        r = sd_device_get_syspath(pci_dev, &pci_syspath);
+        if (r < 0)
+                return r;
+
+        pci_physfn_syspath = strjoina(pci_syspath, "/physfn");
+        r = sd_device_new_from_syspath(&pci_physfn_dev, pci_physfn_syspath);
```

Comment:
we should probably add `sd_device_new_child(sd_device *d, const char *suffix)` or so, which combines the sd_device_get_syspath() + path_join() + sd_device_new_from_syspath() into one. I am pretty sure this is a pretty common pattern in our code...